drivers/serial/uart_native_posix.c

```diff
@@ -184,6 +186,9 @@ static int open_tty(struct native_uart_status *driver_data,
 	posix_print_trace("%s connected to pseudotty: %s\n",
 			  uart_name, slave_pty_name);
 
+	if (wait_pts) {
+		close(open(slave_pty_name, O_RDWR | O_NOCTTY));
```

Comment:
Exactly, will add some comment.

Reply:
Thanks for adding the comment, could we clarify it a little more? Perhaps something like "This trick sets the HUP flag on the tty master, making it possible to detect a client connection using poll. The connection of the client would cause the HUP flag to be cleared, and in turn set again at disconnect." ?

Reply:
Perfectly clear. I'll do that. Thank you.